frontend/controller/actions/identity.js

```diff
@@ -232,6 +235,7 @@ export default (sbp('sbp/selectors/register', {
           ...publishOptions,
           headers: {
             ...publishOptions?.headers,
+            'shelter-deletion-token': deletionToken,
```

Comment:
Would it make more sense, instead of sending the raw deletion token here, to send a hash of it instead, and then reveal the value to the server when we actually want to delete it?

Reply:
Do you have some scenario in mind? I mean, maybe, but it complicates things and I'm not exactly sure what this is supposed to address.

We discussed this implementation at some point. The gist of it is that files have a deletion token that is server-generated. For contracts, we don't have that (we could, but we don't), and the reason to have it be client-generated in this case is to save a few roundtrips (the token is stored in the contract).

If we do the hash approach, then we need to handle that especially (compared to files) and I don't see what function it'd serve, since the server is capable of deleting data without knowing the token.

Reply:
I suppose if this happens over an insecure connection it could be sniffed and used to cause DoS via deletion

Reply:
Maybe don't use insecure connections? We send this in the clear for files (server to client)

Reply:
Sure, but why risk leaving it like this? It should be a minor change to eliminate this potential.

Reply:
> If you feel strongly about this, we can maybe also make it a client-generated value for files and provide a hash

Yeah I don't have time right now to look at how files work, maybe we can look at it on the call tomorrow. If it's also a minor change then go for it

Reply:
Everything is a hash now. I don't think the insecure connection thing is that much of an issue, but I guess a more probable scenario is the server DB being leaked. This would at least make it harder to use that to start deleting things.